lib/Dialect/AIEVec/Transforms/ConvertVectorToAIEVec.cpp

```diff
@@ -284,51 +310,46 @@ struct ConvertMulToAIEVecMulElemOpPattern
-      auto rhs = dyn_cast<arith::ExtSIOp>(mulOp->getOperand(1).getDefiningOp());
-
-      if (!lhs || !rhs)
-        return failure();
-
-      auto lval = lhs->getOperand(0);
-      auto rval = rhs->getOperand(0);
-
-      VectorType lSrcType = cast<VectorType>(lval.getType());
-      VectorType rSrcType = cast<VectorType>(rval.getType());
-
-      unsigned lBitWidth = lSrcType.getElementType().getIntOrFloatBitWidth();
-      unsigned rBitWidth = rSrcType.getElementType().getIntOrFloatBitWidth();
-
-      if (lBitWidth != 8 || rBitWidth != 8)
-        return failure();
-
-      Type accType = getVectorOpDestType(lSrcType, /*AIEML =*/true);
-      VectorType vecType =
-          createVectorType(512 / lBitWidth, lSrcType.getElementType());
-
-      auto zeroConstOp = rewriter.create<arith::ConstantOp>(
-          lhs.getLoc(), rewriter.getIntegerAttr(lSrcType.getElementType(), 0));
-
-      auto broadcastZeroOp = rewriter.create<aievec::BroadcastScalarOp>(
-          lhs.getLoc(), vecType, zeroConstOp->getResult(0));
-      auto extOp = rewriter.create<aievec::ExtOp>(
-          lhs.getLoc(), lSrcType, broadcastZeroOp.getResult(), 0);
-
-      SmallVector<Value> lSources = {lval, extOp->getResult(0)};
-      SmallVector<Value> rSources = {rval, extOp->getResult(0)};
-      auto lConcatOp =
-          rewriter.create<aievec::ConcatOp>(lhs.getLoc(), vecType, lSources);
-      auto rConcatOp =
-          rewriter.create<aievec::ConcatOp>(rhs.getLoc(), vecType, rSources);
-
-      auto mulElemOp = rewriter.create<aievec::MulElemOp>(
-          mulOp.getLoc(), accType, lConcatOp->getResult(0),
-          rConcatOp->getResult(0));
-      rewriter.replaceOpWithNewOp<aievec::CastOp>(
-          mulOp, resultType, mulElemOp.getResult(), /*isResAcc*/ false);
+    if (laneSize == 32 && (resultElWidth == 32 || resultElWidth == 8)) {
+      if (resultElWidth == 32) {
+        auto lhs =
+            dyn_cast<arith::ExtSIOp>(mulOp->getOperand(0).getDefiningOp());
+        auto rhs =
+            dyn_cast<arith::ExtSIOp>(mulOp->getOperand(1).getDefiningOp());
+
+        if (!lhs || !rhs)
+          return failure();
+
+        auto lval = lhs->getOperand(0);
+        auto rval = rhs->getOperand(0);
+
+        VectorType lSrcType = cast<VectorType>(lval.getType());
+        VectorType rSrcType = cast<VectorType>(rval.getType());
+
+        unsigned lBitWidth = lSrcType.getElementType().getIntOrFloatBitWidth();
+        unsigned rBitWidth = rSrcType.getElementType().getIntOrFloatBitWidth();
+
+        if (lBitWidth != 8 || rBitWidth != 8)
+          return failure();
+
+        auto mulElemOp = createMulElemOp(rewriter, lval, rval, lSrcType,
+                                         lBitWidth, mulOp->getLoc());
+        rewriter.replaceOpWithNewOp<aievec::CastOp>(
+            mulOp, resultType, mulElemOp.getResult(), /*isResAcc*/ false);
+      } else {
+        auto lval = mulOp->getOperand(0);
+        auto rval = mulOp->getOperand(1);
```

Comment:
Same here, use `.getLhs()` and `.getRhs()`. You also want to use the adaptor here.